src/main/java/de/uni_tuebingen/ub/nppm/db/AbstractBase.java

```diff
@@ -260,10 +261,46 @@ protected static String buildAndConditions(Map<String, String> andConditions) {
     //Hilsfunktion zum setzen der Parameter
     protected static void registerAndConditions(Map<String, String> andConditions, NativeQuery query) {
         for (Map.Entry<String, String> andCondition : andConditions.entrySet()) {
-            query.setParameter(andCondition.getKey(), andCondition.getValue());
+
+            if (andCondition.getKey().equals("QuelleBisJahrhundert") || andCondition.getKey().equals("QuelleVonJahrhundert") || andCondition.getKey().equals("VonJahrhundert") || andCondition.getKey().equals("BisJahrhundert") ) {
+                query.setParameter(andCondition.getKey(), andCondition.getValue(), StringType.INSTANCE);
+            } else {
+                query.setParameter(andCondition.getKey(), andCondition.getValue());
+            }
         }
     }
 
+     public static void updateMap(String table, Map<String, String> attributesAndValues, Map<String, String> andConditions) throws Exception {
```

Comment:
You can rename this function to "update" as well & simply use function overloading this way.

Reply:
done: look at commit: improvement of function update